modules/storages/app/views/storages/admin/storages/_form.html.erb

```diff
@@ -0,0 +1,58 @@
+<%#-- copyright
+OpenProject is an open source project management software.
+Copyright (C) 2012-2022 the OpenProject GmbH
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License version 3.
+
+OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+Copyright (C) 2006-2013 Jean-Philippe Lang
+Copyright (C) 2010-2013 the ChiliProject Team
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+See COPYRIGHT and LICENSE files for more details.
+
+++#%>
+
+<!-- Included form for create and edit operations, please see the controller for comments -->
+<%= error_messages_for_contract @object, @errors %>
+
+<section class="form--section">
+  <div class="form--field -required">
+    <%= f.select :provider_type,
+                 ::Storages::Storage::PROVIDER_TYPES.map { |provider_type| [I18n.t("storages.provider_types.#{provider_type}"), provider_type] },
+                 selected: 'nextcloud',
```

Comment:
This is somewhat of a magic string the way it is implemented right now. 

A way out of this would be to turn `Storages::Storage#provider_type` into an [enum](https://api.rubyonrails.org/v6.1/classes/ActiveRecord/Enum.html). If the column needs to be kept a string, that also [seems to be possible](https://stackoverflow.com/a/32939934/3206935)

Reply:
> This is somewhat of a magic string the way it is implemented right now.
> 
> A way out of this would be to turn `Storages::Storage#provider_type` into an [enum](https://api.rubyonrails.org/v6.1/classes/ActiveRecord/Enum.html). If the column needs to be kept a string, that also [seems to be possible](https://stackoverflow.com/a/32939934/3206935)

I do not quite like ruby enums in this case. I used another approach in 9bc93a91: 
https://github.com/opf/openproject/commit/9bc93a91#diff-3efae49d897521101f17ed027f06187369ab0248671bdc1574bca588dad618cbL55-R57

Reply:
Looks good @cbliard

Reply:
Done in 65b6d4aead. Provider choice is disabled for edition, and when there is only 1 provider.